homeassistant/components/hunterdouglas_powerview/select.py

```diff
@@ -0,0 +1,122 @@
+from .model import PowerviewDeviceInfo, PowerviewEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PowerviewSelectDescriptionMixin:
+    """Mixin to describe a select entity."""
+
+    current_fn: Callable[[BaseShade], Any]
+    select_fn: Callable[[BaseShade, str], Coroutine[Any, Any, bool]]
+
+
+@dataclass
+class PowerviewSelectDescription(
+    SelectEntityDescription, PowerviewSelectDescriptionMixin
+):
+    """A class that describes select entities."""
+
+    entity_category: EntityCategory = EntityCategory.CONFIG
+
+
+DROPDOWNS: Final = [
+    PowerviewSelectDescription(
+        key="powersource",
+        name="Power Source",
+        icon="mdi:power-plug-outline",
+        current_fn=lambda shade: POWER_SUPPLY_TYPE_MAP.get(
+            shade.raw_data.get(ATTR_BATTERY_KIND)
+        ),
+        options=list(POWER_SUPPLY_TYPE_MAP.values()),
+        select_fn=lambda shade, option: shade.set_power_source(
+            POWER_SUPPLY_TYPE_REVERSE_MAP.get(option)
+        ),
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the hunter douglas select entities."""
+
+    pv_entry: PowerviewEntryData = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+    for raw_shade in pv_entry.shade_data.values():
+        shade: BaseShade = PvShade(raw_shade, pv_entry.api)
+        if SHADE_BATTERY_LEVEL not in shade.raw_data:
+            continue
+        name_before_refresh = shade.name
+        room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
+        room_name = pv_entry.room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
+
+        for description in DROPDOWNS:
+            entities.append(
+                PowerViewSelect(
+                    pv_entry.coordinator,
+                    pv_entry.device_info,
+                    room_name,
+                    shade,
+                    name_before_refresh,
+                    description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class PowerViewSelect(ShadeEntity, SelectEntity):
+    """Representation of a select entity."""
+
+    def __init__(
+        self,
+        coordinator: PowerviewShadeUpdateCoordinator,
+        device_info: PowerviewDeviceInfo,
+        room_name: str,
+        shade: BaseShade,
+        name: str,
+        description: PowerviewSelectDescription,
+    ) -> None:
+        """Initialize the select entity."""
+        super().__init__(coordinator, device_info, room_name, shade, name)
+        self.entity_description: PowerviewSelectDescription = description
+        self._attr_name = f"{self._shade_name} {description.name}"
+        self._attr_unique_id = f"{self._attr_unique_id}_{description.key}"
+
+    @property
+    def current_option(self) -> str:
+        """Return the selected entity option to represent the entity state."""
+        return str(self.entity_description.current_fn(self._shade))
```

Comment:
Should this be a `cast` instead of `str`?  I think its already a `str`

Reply:
casting to str was only because it is also possible for "unassigned" if the user has not set the value in the app - casting an empty str resulted in the dropdown providing no selection. Powerview app treats 'unassigned' as battery wand but i thought it best to not assign a type if they are not set

Reply:
so the 4 values i have seen in the batterykind are '1','2','3','unassigned'
I havent mapped unassigned as i coded to treat anything not 1/2/3 as not set

Reply:
Not sure how it would behave in the UI but we usually use None to represent no value

Reply:
pushed a change to default to None and removed the cast